packages/tupaia-web-server/src/routes/MapOverlaysRoute.ts

```diff
@@ -141,6 +156,11 @@ export class MapOverlaysRoute extends Route<MapOverlaysRequest> {
         (relation: MapOverlayGroupRelation) => {
           if (relation.child_type === MAP_OVERLAY_CHILD_TYPE) {
             const overlay = overlaysById[relation.child_id];
+
+            // If the measure level is found in the ancestor types, that means the currently selected entity is a descendant of the measure level entity, so there will be no data to display. In this case, the overlay should be disabled.
+            const isDisabled = overlay.config.measureLevel
```

Comment:
Nice - I think this is a great solution in the end. It took 2 years but we finally figured it out 😆